File: Bank Account Application (Java)/BankAccount.java
import java.util.*;

class Main {
  public static void main(String[] args) {
//Constructors
    
    Scanner input = new Scanner(System.in);

//Greeting User + Asking how much do they currently have in both accounts
    System.out.println("Hello User and welcome to your checking and savings account program!");
    System.out.println("To start, would you like to begin?\n>>Yes\n>>No");
    String userChoice = input.nextLine().toLowerCase();
    
    
if(userChoice.equals("yes")){
  BankAccount Checking = new BankAccount("Checking", 0);
  BankAccount Savings = new BankAccount("Savings",0);
  
  
  System.out.println("To start, how much do you have in your checking account?");
  int checkingBalance = Integer.parseInt(input.nextLine());
  
  System.out.println("How much do you have in your savings account?");
  int savingsBalance = Integer.parseInt(input.nextLine());
  
  if(savingsBalance >= 0 && checkingBalance >= 0){  
  Checking.setBalance(checkingBalance);
  Savings.setBalance(savingsBalance);
  
  System.out.println("How much would you like to transfer from you Checking to your Savings?");
  int transferAmount = Integer.parseInt(input.nextLine());
  while(true){
    
    
    
    if(transferAmount >= 0){
      Checking.transferMoney(Savings, transferAmount);
      System.out.println("Your new Checking balance is: $"+Checking.getBalance());
      System.out.println("Your new Savings balance is: $"+Savings.getBalance());
      
      System.out.println("Would you like to continue?\n>>Yes\n>>No");
      String shouldContinue = input.nextLine().toLowerCase();

      if(shouldContinue.equals("yes")||shouldContinue.equals("no")){
        if(shouldContinue.equals("yes")){System.out.println("Okay, how much would you like to transfer?");transferAmount = Integer.parseInt(input.nextLine());}
        if(shouldContinue.equals("no")){System.out.println("Okay, have a nice day!"); break;}
      }else{System.out.println("Sorry, the decision you have entered is not understood, please try again.");}
    }else{System.out.println("Sorry, the program does not allow negative values. Have a nice day!");break;}
  }
  }else{System.out.println("Sorry, the program does not allow negative values. Have a nice day!");}
/**/}else{System.out.println("Okay, have a nice day!");}
  }
}
class BankAccount{
  protected String accountType;
  protected int accountBalance = 0;
//Setting up Constructor
  public BankAccount(String AccountType, int AccountBalance){
    accountType = AccountType;
    accountBalance = AccountBalance;
  }
//Getter + Setter for Balance
  public int getBalance(){return this.accountBalance;}
  public void setBalance(int AccountBalance){accountBalance = AccountBalance;}
  
//Getter + Setter for AccountType
  public String getAccountType(){return this.accountType;}
  public void setAccountType(String AccountType){accountType = AccountType;}

//Transfering from Checking to Savings
  public void transferMoney( BankAccount bank, int amount){
    if(amount <= accountBalance){
      this.accountBalance -= amount;
      bank.accountBalance += amount;
    }else{System.out.println("Sorry, your transaction was denied as you do not have enough to transfer.");}
  }
}

File: Bank Account Application (C#)/BankAccount.cs
using System;
					
public class Program
{
	public static void Main()
	{
		string shouldcontinue = "Yes";
		string userchoice = "Yes";
		
		//Greeting User + Asking if the User wants to use the program
		Console.WriteLine("Hello! Welcome to your checking and savings account program!");
		Console.WriteLine("Do you want to begin? Yes or No");
		userchoice = Console.ReadLine();
		
		if(userchoice == "Yes")
		{
			BankAccount Checking = new BankAccount("Checking", 0);
			BankAccount Saving = new BankAccount("Saving", 0);
			
			
			//Getting + Setting Checking Balance
			Console.WriteLine("\nOkay, to start off, how much is currently in your Checking Account?");
			decimal CheckingBalance = Decimal.Parse(Console.ReadLine());
			
			
			Console.WriteLine("\nOkay,now how much is currently in your Saving Account?");
			decimal SavingBalance = Decimal.Parse(Console.ReadLine());
			
			while(shouldcontinue == "Yes")
			{	
			Checking.setBalance(CheckingBalance);
			Saving.setBalance(SavingBalance);
				//Transfering Money From Checking To Savings
				Console.WriteLine("\nOkay now how much do you want to save?");
				decimal TransferAmount = Decimal.Parse(Console.ReadLine());
				
				//Stating New Checking + Saving Balance
				//decimal SavingBalance = Saving.getBalance();
				Transfer.SaveMoney(ref CheckingBalance, ref SavingBalance, TransferAmount);
				
				//Ask User If Need Another Transaction
				Console.WriteLine("\nDo you want to make another transaction? Yes or No");
				shouldcontinue = Console.ReadLine();
			}
			Console.WriteLine("Okay, have a nice day!");
			
		}
		else
		{
			Console.WriteLine("Okay, have a nice day!");
		}
	
	}
	class BankAccount
	{
		private string accounttype;
		private decimal balance;
		
		public BankAccount(string _accounttype, decimal _balance)
		{
			this.accounttype = _accounttype;
			this.balance = _balance;
		}
		
		//Getter + Setter for AccountType
		public string getAccoutType(){return this.accounttype;}
		public void setAccountType(string account){this.accounttype = account;}
		
		//Getter + Setter for balance
		public decimal getBalance()
		{return this.balance;}
		public void setBalance(decimal _balance){this.balance = _balance;}
	}
	
	class Transfer 
	{	
		public static void SaveMoney(ref decimal Checking, ref decimal Saving, decimal transferAmount) 
		{
			if(transferAmount <= Checking)
			{
				Checking = Checking - transferAmount;
				Console.WriteLine("\nOkay, your updated Checking balance is: $" + Checking + ".");
					
				Saving = Saving + transferAmount;
				Console.WriteLine("Here is your update Saving balance is: $" + Saving + ".");
			} 
			else
			{
				Console.WriteLine("Sorry, your transfer request has been denied.");
				Console.WriteLine("You tried to transfer more money into you Saving than you had in your Checking, resulting in Checking balance becoming negative.");
			}
		}
	}
}
